drivers/netdev_layer/netdev_layer.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+#include "net/netdev.h"
+#include "assert.h"
+
+netdev_t *netdev_add_layer(netdev_t *top, netdev_t *netdev)
+{
+    assert(top != NULL);
+    netdev->lower = top;
+    top->context = netdev;
+    return netdev;
+}
+
+int netdev_init_pass(netdev_t *dev)
```

Comment:
Here and below: I'm wondering if we really do need all this functions. They are basically just wrappers for existing functions and just blow up netdev's rather lean interface. At least I would make them `inline` and maybe even put them in `net/netdev/layered.h` or something like that.

Reply:
The reason for these functions is that with the current implementation it is not allowed to have `NULL` values in a netdev_driver struct. So an intermediate netdev layer that doesn't implement all functions can use these to "fill up" the unused functions and simply pass the request to the next layer. Inlining them is thus not useful (I think) as they are only used as function pointers. For example the [netdev_eth code](https://github.com/bergzand/RIOT/blob/wip/netdev_layered_l2total/drivers/netdev_eth/netdev_eth.c#L36) I included here previously, only needs the get and set methods.

Removing these function and allowing `NULL` means that a layer needs to check recursively if any of the next layers implements the called function and call it if it exists. This IMHO makes it a lot more complicated.

Reply:
Ah, I see! Then how about my proposal to moving them to `net/netdev/layered.h` or something similar?

Reply:
No problem, I can move them together with the `netdev_add_layer` function (which can be inlined) to `net/netdev/layered.h and remove the declarations from netdev.h.